joomla/joomla-cms
# Add bootstrap tooltip extended: 4 extra positions and RTL support

This PR introduces bootstrap-tooltip-extended.js to add 4 positions and RTL support for BS tooltips.

It follows discussion about tooltip placement issue when inside modals: https://github.com/joomla/joomla-cms/pull/10100#issuecomment-215035558
#### Summary of Changes
- New <code>JHtml::_('bootstrap.tooltipExtended');</code> method  to load bootstrap-tooltip-extended.js to get 4 extra positions : <code>top-left</code>, <code>top-right</code>, <code>bottom-left</code> and <code>bottom-right</code> (https://github.com/cyrilreze/bootstrap-tooltip-extended)
- ~~Add tooltip-extended arrow css styles in <code>bootstrap-extended.less</code>~~
- Define a default <code>data-placement="auto-dir top-left"</code> (if data-placement is undefined) for class <code>.modalTooltip</code> (handler class for BS tooltip when inside a modal-body).
- In the same time, remove previously changes for Batch modals, when adding tooltip container in PR #9944 and #9379 (this is not needed anymore since the container is now set directly in modal layout main, PR #10100)
#### Testing Instructions
- Apply Patch and empty cache
- Open any modals not using iframe: Batch, Copy Template...
- Check if tooltip placement is top-left (as shown in screenshot below)
- Change language to a RTL language (eg. Arabic)
- Then open again the same modals as in LTR language : the position of tooltip is reversed (top-left becomes top-right)
#### Screenshots

(eg. batch modules)

BEFORE PATCH
![capture d ecran 2016-05-05 a 15 50 02](https://cloud.githubusercontent.com/assets/2385058/15045115/27bab328-12d9-11e6-89fb-cd8b2886e9bb.png)

AFTER PATCH
**Left-to-right (LTR) language direction :** 
![capture d ecran 2016-05-04 a 17 01 16](https://cloud.githubusercontent.com/assets/2385058/15018477/f5d3af50-1219-11e6-95ee-57614c8dfd4f.png)

**Right-to-left (RTL) language direction :** 
![capture d ecran 2016-05-04 a 17 26 29](https://cloud.githubusercontent.com/assets/2385058/15020204/54e68722-1221-11e6-9a15-3b1a9616a492.png)
